src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmCompanionExtension/src/extension.ts

```diff
@@ -0,0 +1,80 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { spawn } from 'child_process';
+import { acquireDotnetInstall } from './acquireDotnetInstall';
+import { getAvailablePort } from "./getAvailablePort";
+
+let launchDebugProxy: vscode.Disposable;
+let killDebugProxy: vscode.Disposable;
+
+export function activate(context: vscode.ExtensionContext) {
+    const outputChannel = vscode.window.createOutputChannel("Blazor WASM Debug Proxy");
```

Comment:
It's interesting that we only get this in Codespaces scenarios but none other. Are there logs going on in the server machine that we should be forwarding?

Reply:
> It's interesting that we only get this in Codespaces scenarios but none other. Are there logs going on in the server machine that we should be forwarding?

We get this in all scenarios. We're defaulting to launching the debug proxy in all scenarios as seen [here](https://github.com/dotnet/aspnetcore-tooling/blob/safia/remote-dbg/src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts#L26-L32).

Reply:
Ah I think that was something I missed. SO in VSCode scenarios we're always going to be the one booting the proxy now, even in local scenarios?

Reply:
Correct. I mentioned this in another comment thread but since we launch the debug proxy independently, we no longer have to wait for the users dev server to launch before being able to initialize the debugger.